editoast/src/views/search.rs

```diff
@@ -333,10 +338,31 @@ struct SearchDBResult {
     )
 )]
 async fn search(
-    Query(query_params): Query<PaginationQueryParam>,
     State(db_pool): State<DbConnectionPoolV2>,
+    Extension(authorizer): AuthorizerExt,
+    Query(query_params): Query<PaginationQueryParam>,
     Json(SearchPayload { object, query, dry }): Json<SearchPayload>,
 ) -> Result<impl IntoResponse> {
+    let mut roles_to_check = HashSet::new();
+    match object.as_str() {
+        "track" | "operationalpoint" | "signal" => roles_to_check.insert(BuiltinRole::InfraRead),
+        "project" | "study" | "scenario" => roles_to_check.insert(BuiltinRole::OpsRead),
+        &_ => {
+            return Err(SearchApiError::ObjectType {
+                object_type: object.to_owned(),
+            }
+            .into())
+        }
+    };
```

Comment:
I think you can write directly:

```suggestion
    let roles: &[BuiltinRole] = match object.as_str() {
        "track" | "operationalpoint" | "signal" => &[BuiltinRole::InfraRead],
        "project" | "study" | "scenario" => &[BuiltinRole::OpsRead],
        _ => {
            return Err(SearchApiError::ObjectType {
                object_type: object.to_owned(),
            }
            .into())
        }
    };
```